sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp

```diff
@@ -259,20 +261,230 @@ AttestationAdministrationClient::ResetAttestationPolicy(
+      Models::_detail::ModifyPolicyCertificatesResult,
+      ModifyPolicyCertificatesResultSerializer>(responseToken);
+
+  // Validate the token returned by the service. Use the cached attestation signers in the
+  // validation.
+  std::vector<AttestationSigner> const& signers = GetAttestationSigners(context);
+  resultToken.ValidateToken(
+      options.TokenValidationOptions ? *options.TokenValidationOptions
+                                     : this->m_tokenValidationOptions,
+      signers);
+
+  // Extract the underlying policy token from the response.
+  auto internalResult
+      = static_cast<AttestationToken<Models::_detail::ModifyPolicyCertificatesResult>>(resultToken)
+            .Body;
+
+  Models::PolicyCertificateModificationResult returnValue;
+  if (internalResult.CertificateResolution)
   {
-    returnedResult.PolicyTokenHash = Base64Url::Base64UrlDecode(*internalResult.PolicyTokenHash);
+    returnValue.CertificateModification
+        = Models::PolicyCertificateModification(*internalResult.CertificateResolution);
+  }
+  if (internalResult.CertificateThumbprint)
+  {
+    returnValue.CertificateThumbprint = (*internalResult.CertificateThumbprint);
   }
 
   // Construct a token whose body is the policy result, but whose token is the response from the
   // service.
-  auto returnedToken
-      = AttestationTokenInternal<Models::PolicyResult>(responseToken, returnedResult);
-  return Response<AttestationToken<Models::PolicyResult>>(returnedToken, std::move(response));
+  auto returnedToken = AttestationTokenInternal<Models::PolicyCertificateModificationResult>(
+      responseToken, returnValue);
+  return Response<AttestationToken<Models::PolicyCertificateModificationResult>>(
+      returnedToken, std::move(response));
+}
+
+Azure::Response<Models::AttestationToken<Models::PolicyCertificateModificationResult>>
+AttestationAdministrationClient::RemovePolicyManagementCertificate(
+    std::string const& pemEncodedX509CertificateToAdd,
+    AttestationSigningKey const& existingSigningKey,
+    AddPolicyManagementCertificatesOptions const& options,
+    Azure::Core::Context const& context) const
+{
+  // Calculate a signed (or unsigned) attestation policy token to send to the service.
+  // Embed the encoded policy in the StoredAttestationPolicy.
+  auto x5cToAdd(Cryptography::ImportX509Certificate(pemEncodedX509CertificateToAdd));
+
+  // Create a JWK to add to the body.
+  JsonWebKey jwkToSend;
+  jwkToSend.kty = x5cToAdd->GetKeyType();
+  jwkToSend.x5c = std::vector<std::string>();
+  jwkToSend.x5c->push_back(x5cToAdd->ExportAsBase64());
+
+  PolicyCertificateManagementBody bodyToSend{jwkToSend};
+
+  auto internalTokenToSend(
+      AttestationTokenInternal<
+          PolicyCertificateManagementBody,
+          PolicyCertificateManagementBodySerializer>::CreateToken(bodyToSend, existingSigningKey));
+
+  auto tokenToSend(
+      static_cast<AttestationToken<PolicyCertificateManagementBody>>(internalTokenToSend));
```

Comment:
This code looks very similar to the code above.  Would it make sense to re-use some of it with some function?